File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0){
            return false;
        }
        String s = String.valueOf(x);
        int i = 0;
        int j = s.length() - 1;
        
        while(i <= j){
            if (s.charAt(i) == s.charAt(j)){
                i += 1;
                j -= 1;
            }
            else return false;
        }
        return true;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) return false;
        long long num = x;
        long long rev = 0;
        while (num){
            rev = rev*10 + num%10;
            num /= 10;
        }
        if (rev == x) return true;
        return false;
        
    }
};